### PR TITLE
add com.qq.QQ

### PR DESCRIPTION
**Pull request by chenzhiwei**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.qq.QQ.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+separate-locales: false
+
+finish-args:
+    - '--share=ipc'
+    - '--share=network'
+    - '--socket=x11'
+    - '--socket=pulseaudio'
+    - '--device=dri'
+    - '--talk-name=org.freedesktop.Notifications'
+    - '--own-name=org.kde.*'
```
Comment (hfiguiere) [inline com.qq.QQ.yml:17]: this is a very broad `own-name`.

File: com.qq.QQ.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    - '--own-name=org.mpris.MediaPlayer2.chromium.*'
```
Comment (hfiguiere) [inline com.qq.QQ.yml:18]: that's chromium's.

File: com.qq.QQ.desktop [hunk 3]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=QQ
+Exec=/opt/QQ/qq %U
```
Comment (hfiguiere) [inline com.qq.QQ.desktop:3]: this is incorrect. there is no `/opt` available.

File: com.qq.QQ.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.qq.QQ</id>
+  <provides><id>com.qq.QQ.desktop</id></provides>
```
Comment (hfiguiere) [inline com.qq.QQ.metainfo.xml:4]:
```suggestion
```

File: com.qq.QQ.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
+app-id: com.qq.QQ
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '22.08'
+command: qq %U
```
Comment (hfiguiere) [inline com.qq.QQ.yml:7]: this is an invalid command.

File: com.qq.QQ.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+build-options:
+    strip: false
+    no-debuginfo: true
+
+modules:
+    -
+        name: qq
+        sources:
+            -
+                type: script
```
Comment (hfiguiere) [inline com.qq.QQ.yml:29]:
```suggestion
            - type: script
```

File: com.qq.QQ.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                commands:
+                    - export TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"
+                    - exec zypak-wrapper /app/extra/QQ/qq "$@"
+                dest-filename: qq.sh
+            -
+                type: script
```
Comment (hfiguiere) [inline com.qq.QQ.yml:35]:
```suggestion
            - type: script
```

File: com.qq.QQ.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                commands:
+                    - ar x qq.deb
+                    - tar -xf data.tar.xz
+                    - mv opt/QQ .
+                    - rm -rf opt usr control.tar.gz data.tar.xz debian-binary
+                dest-filename: apply_extra
+            -
+                type: file
```
Comment (hfiguiere) [inline com.qq.QQ.yml:43]:
```suggestion
            - type: file
```

File: com.qq.QQ.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                path: com.qq.QQ.desktop
+            -
+                type: file
```
Comment (hfiguiere) [inline com.qq.QQ.yml:46]:
```suggestion
            - type: file
```

File: com.qq.QQ.yml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                path: com.qq.QQ.png
+            -
+                type: file
```
Comment (hfiguiere) [inline com.qq.QQ.yml:49]:
```suggestion
            - type: file
```

File: com.qq.QQ.yml [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+                path: com.qq.QQ.metainfo.xml
+            -
+                type: extra-data
```
Comment (hfiguiere) [inline com.qq.QQ.yml:52]:
```suggestion
            - type: extra-data
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

File: com.qq.QQ.yml (hunk 5 above)
Comment (chenzhiwei) [reply com.qq.QQ.yml:7]: thank you, just pushed another patchset and addressed all your comments.

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

File: com.qq.QQ.yml [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+            - tar -xf data.tar.xz
+            - mv opt/QQ .
+            - rm -rf opt usr control.tar.gz data.tar.xz debian-binary
+        dest-filename: apply_extra
+      - type: file
+        path: com.qq.QQ.desktop
+      - type: file
+        path: com.qq.QQ.png
+      - type: file
+        path: com.qq.QQ.metainfo.xml
```
Comment (hfiguiere) [inline com.qq.QQ.yml:53]: this file is never installed.

File: com.qq.QQ.yml (hunk 5 above)
Comment (hfiguiere) [reply com.qq.QQ.yml:7]: this remain invalid. There is NO argument.
```suggestion
command: qq
```

File: com.qq.QQ.yml [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  - '--filesystem=xdg-download'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
+  - '--filesystem=xdg-run/pipewire-0'
+
+modules:
+  - name: qq
+    buildsystem: simple
+    build-commands:
+      - install apply_extra ${FLATPAK_DEST}/bin
```
Comment (hfiguiere) [inline com.qq.QQ.yml:28]:
```suggestion
      - install -Dm755 apply_extra -t ${FLATPAK_DEST}/bin
```

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

File: com.qq.QQ.metainfo.xml [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    </screenshot>
+  </screenshots>
+  <categories>
+    <category>InstantMessaging</category>
+    <category>Network</category>
+  </categories>
+  <releases>
+    <release version="2.0.1" date="2022-12-08"/>
+  </releases>
+  <content_rating type="oars-1.0">
```
Comment (hfiguiere) [inline com.qq.QQ.metainfo.xml:30]:
```suggestion
  <content_rating type="oars-1.1">
```

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by catsout** [comment]
@chenzhiwei
Please use desktop file and icon from deb, example(`apply_extra` script):  
```
          - mv usr export && rm -r export/share/icons
          - install -Dm644 opt/QQ/resources/app/512x512.png export/share/icons/hicolor/512x512/apps/qq.png
          - FLATPAK_ID=com.qq.im.linuxqq
          - desktop-file-edit 
            --set-icon="$FLATPAK_ID" 
            --set-key="Exec" --set-value="qq %U" 
            export/share/applications/*.desktop
          - rename "qq" "$FLATPAK_ID" 
            export/share/icons/hicolor/*/apps/*.png 
            export/share/applications/*.desktop
```

**Comment by catsout** [comment]
You need to use 21.08 version of freedesktop runtime.  
Otherwise, it will fail with
image

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
@catsout Thank you for you valuable comments. Just invited you to join this repo. When you accept, I will give you admin permission.

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by catsout** [comment]
@chenzhiwei 
You need to add `<icon type="remote">https://xxx</icon>` to metainfo.

**Comment by catsout** [comment]
QQ icon source:  
https://wiki.connect.qq.com/%E8%A7%86%E8%A7%89%E7%B4%A0%E6%9D%90%E4%B8%8B%E8%BD%BD

File: com.qq.QQ.metainfo.xml [hunk 15]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.qq.QQ</id>
+  <name>Linux QQ</name>
+  <project_license>LicenseRef-proprietary=https://rule.tencent.com/rule/preview/46a15f24-e42c-4cb6-a308-2347139b1201</project_license>
+  <developer_name>Tencent</developer_name>
```
Comment (catsout) [inline com.qq.QQ.metainfo.xml:6]:
```suggestion
  <developer_name>Tencent</developer_name>
  <icon type="remote">https://wiki.connect.qq.com/wp-content/uploads/2013/10/03_qq_symbol-1-768x921.png</icon>
```

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

File: com.qq.QQ.yaml [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+  - '--share=network'
+  - '--socket=x11'
+  - '--socket=pulseaudio'
+  - '--device=all'
+  - '--filesystem=xdg-documents:ro'
+  - '--filesystem=xdg-pictures:ro'
+  - '--filesystem=xdg-videos:ro'
+  - '--filesystem=xdg-download'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--talk-name=org.kde.StatusNotifierWatcher'
```
Comment (catsout) [inline com.qq.QQ.yaml:21]:
```suggestion
  - '--talk-name=org.kde.StatusNotifierWatcher'
  - '--system-talk-name=org.freedesktop.login1'
```
fix 
```
Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
Failed to call method: org.freedesktop.login1.Manager.Inhibit: object_path= /org/freedesktop/login1
```

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

File: com.qq.QQ.yaml [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+        dest-filename: qq.sh
+
+      - type: script
+        commands:
+          - ar x qq.deb
+          - tar -xf data.tar.xz
+          - mv opt/QQ .
+          - mv usr export
+          - rm -rf export/share/icons
+          - install -Dm644 opt/QQ/resources/app/512x512.png export/share/icons/hicolor/512x512/apps/qq.png
```
Comment (catsout) [inline com.qq.QQ.yaml:55]:
```suggestion
          - install -Dm644 QQ/resources/app/512x512.png export/share/icons/hicolor/512x512/apps/qq.png
```
As `mv opt/QQ .`, remove `opt`.

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
@hfiguiere @catsout Thank you for you great help, I will continue improving this package.

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
bot, build com.qq.QQ

**Comment by chenzhiwei** [comment]
The new Linux QQ is still in limited testing, will add arm64 when it generally available.

File: com.qq.QQ.yaml [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+        url: https://dldir1.qq.com/qqfile/qq/QQNT/4691a571/QQ-v2.0.1-429_x64.deb
+        sha256: e3aa15ff6ae089b655df3913c6020ca56726bd676995aaf29b313fbd0643ea42
+        size: 108719314
+        # Will enable when available
+        # x-checker-data:
+        #     type: html
+        #     url: https://im.qq.com/linuxqq/download.html
+        #     version-pattern: '(\S*)'
+        #     url-template: 'https://dldir1.qq.com/qqfile/qq/QQNT/4691a571/QQ-(\S*)_x64.deb'
+  - name: kerberos
```
Comment (hfiguiere) [inline com.qq.QQ.yaml:75]: this module should be before the "main" one as it is a dependency.

File: com.qq.QQ.yaml [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+      - type: script
+        commands:
+          - ar x qq.deb
+          - tar -xf data.tar.xz
+          - mv opt/QQ .
+          - mv usr export
+          - rm -rf export/share/doc
+          - rm -rf export/share/icons
+          - install -Dm644 QQ/resources/app/512x512.png export/share/icons/hicolor/512x512/apps/qq.png
+          - FLATPAK_ID=com.qq.QQ
```
Comment (hfiguiere) [inline com.qq.QQ.yaml:57]:
```suggestion
```

FLATPAK_ID is set in the environment. You should never change it.

File: com.qq.QQ.yaml (hunk 18 above)
Comment (barthalion) [reply com.qq.QQ.yaml:75]: Yeah, although it probably doesn't matter much as it repackages the binary.

**Pull request closed without merging**